scripts/tests/test_generate_reference_fixtures.py

```
@@ -0,0 +1,45 @@
+                "operating_system": "Linux",
+                "hardware": "x86_64",
+            }
+        }
+        scaling = [
+            {
+                "build.system": "Linux",
+                "build.processor": "x86_64",
+            }
+        ]
+
+        with self.assertRaisesRegex(RuntimeError, "macOS-arm64-specific"):
```

Ruff (0.15.21)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)